src/data_structures/arguments.rs

```diff
@@ -453,13 +440,17 @@ impl<'a> QuotientCache<'a> {
     pub fn get_or_compute_coset_evals(
         &mut self,
         coset_idx: usize,
-    ) -> CudaResult<&GenericComplexPolynomialStorage<'a, CosetEvaluations>> {
+    ) -> CudaResult<Rc<GenericComplexPolynomialStorage<'a, CosetEvaluations>>> {
```

Comment:
Return value here has a lifetime annotation, but [corresponding return value for ArgumentCache](https://github.com/matter-labs/era-shivini/pull/7/files#diff-855778c86fcd4c999230857a58bb1b4d9c296959171ae02de5ebb4c65cd63661R332) doesnt...is this deliberate?

Reply:
The original code for ArgumentCache had no lifetime annotation either.

Reply:
I don't consider making changes to lifetime annotations to be in scope of this PR, i am keeping the same annotations or lack of annotations as the original code.